src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs

```diff
@@ -105,8 +109,7 @@ public RestoreCommand(IApplicationEnvironment env)
                         Interlocked.Increment(ref successCount);
                     }
                 };
-
-                if (PlatformHelper.IsMono)
+                if (!RestoreInParallel || PlatformHelper.IsMono)
```

Comment:
On mono if you do --parallel it will not work, unless we just remove the option on mono